Stratis.Bitcoin.Features.Consensus/ConsensusStats.cs

```diff
@@ -100,5 +92,12 @@ public void Log()
             benchLogs.AppendLine(this.connectionManager.GetStats());
             this.logger.LogInformation(benchLogs.ToString());
         }
+
+        protected override void OnNextCore(Block value)
```

Comment:
This is interesting. Not sure how I feel about it (btw inheritdoc comment missing i guess). So if i understand this correctly, instead of having there a loop, i.e. something like a timer that would just put these stats on the console, you are now using this signaller so that you are executing within the context of the consensus loop thread. Is it good for anything? 

I mean, I understand that this prevents concurrency problems if there were any (well there were, but with chain, not sure if with consensus loop), but the nature of this thing sounds to me more like the timer (or async loop). 

So now this methods when it gets executed delays consensus from doing its job. Is that a good thing?  I know there is a parallel with my coinview flushing approach, but in that code there really was a concurrency problem with consensus, so although I was not really happy about that I did like this there. 

Doing it here. Well, the performance hit is very very small, so I would OK. Just not sure I like it from the design point of view. 

Not a big deal anyway, just wondered what is the rationale here, if I'm missing something or not.

Reply:
Actually this does not behave differently then before, the consensus loop would have also block to update the stats.

Perhaps we should make it a background loop in a (later PR) or as part of the node bench loop

Reply:
Oh, i can see it now, i was expecting the original was something else. You are right

Reply:
this is done